Activate the 'save' button with 'enter' key in src/libs/tagging.c

In Darktable GTK on Linux, some pop up windows like the 'edit tag' window do not activate the 'Yes/OK/Save' button by default. As a consequence, it is not possible to use the 'enter / return' key to activate/press the button and the mouse must be used to click on it.

A very simple solution would be to activate the default 'confirm/OK/save' button.

src/libs/tagging.c
 _pop_menu_dictionary_edit_tag(GtkWidget *menuitem, dt_lib_module_t *self)
[...]

  // Set 'Enter' key to activate the default button and make 'OK' button the default button
  gtk_dialog_set_activatable(dialog, TRUE);
  GtkWidget *ok_button = gtk_dialog_get_widget_for_response(dialog, Gtk.ResponseType.OK);
  gtk_button_set_can_default(ok_button, TRUE);
  gtk_button_grab_default(ok_button);


When editing lots of tags, it saves precious time.
What do you think ?

Hello, I am trying to compiler using the terminal but I run in to some issues. I have a few questions

> you have previously built darktable, don't forget to first completely remove (rm -R) the build and /opt/darktable directories to avoid conflicting files from different versions.

I understand removing the previous build and do that but what does " _**/opt/darktable directories"**_ mean?

2.) I have g++-12 installed and have removed g++ 11 but when building it says 11 is the version detected.

I have downloaded the stable release 4.4.2, Is that fine? and the below response is from doing `./build.sh`

````

Building directory:  /home/rmknan/Github/darktable/build
Installation prefix: /opt/darktable
Build type:          RelWithDebInfo
Build generator:     Unix Makefiles
Build tasks:         12

```
-- The CXX compiler identification is Clang 14.0.0
-- The C compiler identification is GNU 11.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
  (Required is at least version "4.5")
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindOpenMP.cmake:544 (find_package_handle_standard_args)
  CMakeLists.txt:90 (find_package)
`
```
```

--- a/data/themes/darktable-elegant-grey.css
+++ b/data/themes/darktable-elegant-grey.css
@@ -51,7 +51,7 @@
 @define-color bauhaus_fg_insensitive alpha(@bauhaus_fg, 0.5);
 
 /* GTK Buttons and tabs */
-@define-color button_bg @grey_55;
+@define-color button_bg shade(@grey_55,0.95);
 @define-color button_hover_bg @grey_70;
 @define-color button_hover_fg @grey_30;
 

--- a/data/themes/darktable.css
+++ b/data/themes/darktable.css
@@ -1669,6 +1669,7 @@ progressbar progress
   ------------------------------------------------------------------*/
 /*** All states options are designed below, active, hover, selected, disabled and check states ***/
 /* Main states */
+.default, /* dialog button that activates with Enter */
 .dt_history_items:active,
 .dt_history_items:checked,
 button:active,
@@ -1682,7 +1683,6 @@ entry:checked,
 filechooser *:selected,
 menuitem:active,
 menuitem:selected,
-messagedialog .default,  /* this show that a button in confirmation action pop-up has the focus (no by default) */
 treeview *:selected:not(check),
 treeview *:checked:not(check),
 #main-histogram .dt_module_btn:checked:not(.rgb_toggle)

--- a/src/common/darktable.c
+++ b/src/common/darktable.c
@@ -1522,7 +1522,7 @@ int dt_init(int argc, char *argv[], const gboolean init_gui, const gboolean load
       not_again = dt_gui_show_standalone_yes_no_dialog
         (_("configuration information"),
          config_info,
-         _("show this information again"), _("understood"));
+         _("_show this information again"), _("_understood"));
 
     if(not_again || (last_configure_version == 0))
       dt_conf_set_int("performance_configuration_version_completed",

--- a/src/common/database.c
+++ b/src/common/database.c
@@ -3093,14 +3093,14 @@ void dt_database_show_error(const dt_database_t *db)
     // clang-format on
 
     gboolean delete_lockfiles = dt_gui_show_standalone_yes_no_dialog(_("error starting darktable"),
-                                        label_text, _("cancel"), _("delete database lock files"));
+                                        label_text, _("_cancel"), _("_delete database lock files"));
 
     if(delete_lockfiles)
     {
       gboolean really_delete_lockfiles =
         dt_gui_show_standalone_yes_no_dialog
         (_("are you sure?"),
-         _("\ndo you really want to delete the lock files?\n"), _("no"), _("yes"));
+         _("\ndo you really want to delete the lock files?\n"), _("_no"), _("_yes"));
       if(really_delete_lockfiles)
       {
         int status = 0;
@@ -3118,14 +3118,14 @@ void dt_database_show_error(const dt_database_t *db)
         if(status==0)
           dt_gui_show_standalone_yes_no_dialog(_("done"),
                                         _("\nsuccessfully deleted the lock files.\nyou can now restart darktable\n"),
-                                        _("ok"), NULL);
+                                        _("_ok"), NULL);
         else
           dt_gui_show_standalone_yes_no_dialog
             (_("error"), g_markup_printf_escaped(
               _("\nat least one file could not be removed.\n"
                 "you may try to manually delete the files <i>data.db.lock</i> and <i>library.db.lock</i>\n"
                 "in folder <a href=\"file:///%s\">%s</a>.\n"), lck_dirname, lck_dirname),
-             _("ok"), NULL);
+             _("_ok"), NULL);
       }
     }
 
@@ -3340,7 +3340,7 @@ void ask_for_upgrade(const gchar *dbname, const gboolean has_gui)
 
   gboolean shall_we_update_the_db =
     dt_gui_show_standalone_yes_no_dialog(_("darktable - schema migration"), label_text,
-                                         _("close darktable"), _("upgrade database"));
+                                         _("_close darktable"), _("_upgrade database"));
 
   g_free(label_text);
 
@@ -3661,12 +3661,9 @@ start:
         dialog = gtk_dialog_new_with_buttons(_("darktable - error opening database"),
                                             NULL,
                                             dflags,
-                                            _("close darktable"),
-                                            GTK_RESPONSE_CLOSE,
-                                            _("attempt restore"),
-                                            GTK_RESPONSE_ACCEPT,
-                                            _("delete database"),
-                                            GTK_RESPONSE_REJECT,
+                                            _("_close darktable"), GTK_RESPONSE_CLOSE,
+                                            _("_attempt restore"), GTK_RESPONSE_ACCEPT,
+                                            _("_delete database"), GTK_RESPONSE_REJECT,
                                             NULL);
         gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
         label_options = _("do you want to close darktable now to manually restore\n"
@@ -3679,10 +3676,8 @@ start:
         dialog = gtk_dialog_new_with_buttons(_("darktable - error opening database"),
                                             NULL,
                                             dflags,
-                                            _("close darktable"),
-                                            GTK_RESPONSE_CLOSE,
-                                            _("delete database"),
-                                            GTK_RESPONSE_REJECT,
+                                            _("_close darktable"), GTK_RESPONSE_CLOSE,
+                                            _("_delete database"), GTK_RESPONSE_REJECT,
                                             NULL);
         gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_CLOSE);
         label_options = _("do you want to close darktable now to manually restore\n"
@@ -3848,12 +3843,9 @@ start:
       dialog = gtk_dialog_new_with_buttons(_("darktable - error opening database"),
                                           NULL,
                                           dflags,
-                                          _("close darktable"),
-                                          GTK_RESPONSE_CLOSE,
-                                          _("attempt restore"),
-                                          GTK_RESPONSE_ACCEPT,
-                                          _("delete database"),
-                                          GTK_RESPONSE_REJECT,
+                                          _("_close darktable"), GTK_RESPONSE_CLOSE,
+                                          _("_attempt restore"), GTK_RESPONSE_ACCEPT,
+                                          _("_delete database"), GTK_RESPONSE_REJECT,
                                           NULL);
       gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
       label_options = _("do you want to close darktable now to manually restore\n"
@@ -3866,10 +3858,8 @@ start:
       dialog = gtk_dialog_new_with_buttons(_("darktable - error opening database"),
                                           NULL,
                                           dflags,
-                                          _("close darktable"),
-                                          GTK_RESPONSE_CLOSE,
-                                          _("delete database"),
-                                          GTK_RESPONSE_REJECT,
+                                          _("_close darktable"), GTK_RESPONSE_CLOSE,
+                                          _("_delete database"), GTK_RESPONSE_REJECT,
                                           NULL);
       gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_CLOSE);
       label_options = _("do you want to close darktable now to manually restore\n"

--- a/src/gui/gtk.c
+++ b/src/gui/gtk.c
@@ -2522,7 +2522,7 @@ gboolean dt_gui_show_standalone_yes_no_dialog(const char *title, const char *mar
 
   if(no_text)
   {
-    button = gtk_button_new_with_label(no_text);
+    button = gtk_button_new_with_mnemonic(no_text);
     result.button_no = button;
     g_signal_connect(G_OBJECT(button), "clicked", G_CALLBACK(_yes_no_button_handler), &result);
     gtk_box_pack_start(GTK_BOX(hbox), button, TRUE, TRUE, 0);
@@ -2530,7 +2530,7 @@ gboolean dt_gui_show_standalone_yes_no_dialog(const char *title, const char *mar
 
   if(yes_text)
   {
-    button = gtk_button_new_with_label(yes_text);
+    button = gtk_button_new_with_mnemonic(yes_text);
     result.button_yes = button;
     g_signal_connect(G_OBJECT(button), "clicked", G_CALLBACK(_yes_no_button_handler), &result);
     gtk_box_pack_start(GTK_BOX(hbox), button, TRUE, TRUE, 0);

--- a/src/gui/presets.c
+++ b/src/gui/presets.c
@@ -558,14 +558,14 @@ static void _presets_show_edit_dialog(dt_gui_presets_edit_dialog_t *g,
   char title[1024];
   snprintf(title, sizeof(title), _("edit `%s' for module `%s'"),
            g->original_name, g->module_name);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons
-    (title, g->parent, GTK_DIALOG_DESTROY_WITH_PARENT | GTK_DIALOG_MODAL,
-     _("_export..."), GTK_RESPONSE_YES,
-     _("delete"), GTK_RESPONSE_REJECT,
-     _("_cancel"), GTK_RESPONSE_CANCEL,
-     _("_ok"), GTK_RESPONSE_OK, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(title, g->parent,
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT | GTK_DIALOG_MODAL,
+                                                  _("_export..."), GTK_RESPONSE_YES,
+                                                  _("_delete"), GTK_RESPONSE_REJECT,
+                                                  _("_cancel"), GTK_RESPONSE_CANCEL,
+                                                  _("_ok"), GTK_RESPONSE_OK, NULL);
   dt_gui_dialog_add_help(GTK_DIALOG(dialog), "preset_dialog");
-  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_OK);
 
   g->dialog = dialog;
 
@@ -579,6 +579,7 @@ static void _presets_show_edit_dialog(dt_gui_presets_edit_dialog_t *g,
 
   g->name = GTK_ENTRY(gtk_entry_new());
   gtk_entry_set_text(g->name, g->original_name);
+  gtk_entry_set_width_chars(g->name, 10 + g_utf8_strlen(title, -1));
   if(allow_name_change)
     gtk_entry_set_activates_default(g->name, TRUE);
   else

--- a/src/gui/styles_dialog.c
+++ b/src/gui/styles_dialog.c
@@ -488,6 +488,13 @@ static gint _g_list_find_module_by_name(gconstpointer a, gconstpointer b)
   return strncmp(((dt_iop_module_t *)a)->op, b, strlen(((dt_iop_module_t *)a)->op));
 }
 
+static void _name_changed(GtkEntry *entry,
+                          GtkDialog *dialog)
+{
+  const gchar *name = gtk_entry_get_text(entry);
+  gtk_dialog_set_response_sensitive(dialog, GTK_RESPONSE_ACCEPT, name && *name);
+}
+
 static void _gui_styles_dialog_run(gboolean edit, const char *name, dt_imgid_t imgid)
 {
   char title[512];
@@ -514,11 +521,12 @@ static void _gui_styles_dialog_run(gboolean edit, const char *name, dt_imgid_t i
   GtkWidget *window = dt_ui_main_window(darktable.gui->ui);
   GtkDialog *dialog = GTK_DIALOG(
       gtk_dialog_new_with_buttons(title, GTK_WINDOW(window), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                   _("select _all"),  GTK_RESPONSE_YES,
+                                  _("select _all"),  GTK_RESPONSE_YES,
                                   _("select _none"), GTK_RESPONSE_NONE,
                                   _("_cancel"), GTK_RESPONSE_REJECT,
-                                 _("_save"), GTK_RESPONSE_ACCEPT, NULL));
+                                  _("_save"), GTK_RESPONSE_ACCEPT, NULL));
   dt_gui_dialog_add_help(dialog, "styles");
+  gtk_dialog_set_default_response(dialog, GTK_RESPONSE_ACCEPT);
 
 #ifdef GDK_WINDOWING_QUARTZ
   dt_osx_disallow_fullscreen(GTK_WIDGET(dialog));
@@ -547,11 +555,15 @@ static void _gui_styles_dialog_run(gboolean edit, const char *name, dt_imgid_t i
   sd->name = gtk_entry_new();
   gtk_entry_set_placeholder_text(GTK_ENTRY(sd->name), _("name"));
   gtk_widget_set_tooltip_text(sd->name, _("enter a name for the new style"));
+  gtk_entry_set_activates_default(GTK_ENTRY(sd->name), TRUE);
+  gtk_dialog_set_response_sensitive(dialog, GTK_RESPONSE_ACCEPT, FALSE);
+  g_signal_connect(sd->name, "changed", G_CALLBACK(_name_changed), dialog);
 
   sd->description = gtk_entry_new();
   gtk_entry_set_placeholder_text(GTK_ENTRY(sd->description), _("description"));
   gtk_widget_set_tooltip_text(sd->description,
                               _("enter a description for the new style, this description is searchable"));
+  gtk_entry_set_activates_default(GTK_ENTRY(sd->description), TRUE);
 
   /*set values*/
   if(edit && name)

--- a/src/libs/collect.c
+++ b/src/libs/collect.c
@@ -2920,8 +2920,9 @@ void _menuitem_preferences(GtkMenuItem *menuitem, dt_lib_module_t *self)
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
   GtkWidget *dialog = gtk_dialog_new_with_buttons(_("collections settings"), GTK_WINDOW(win),
                                                   GTK_DIALOG_DESTROY_WITH_PARENT,
-                                                 _("cancel"), GTK_RESPONSE_NONE,
-                                                 _("save"), GTK_RESPONSE_ACCEPT, NULL);
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_ACCEPT, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
   dt_prefs_init_dialog_collect(dialog);
   g_signal_connect(dialog, "key-press-event", G_CALLBACK(dt_handle_dialog_enter), NULL);
 

--- a/src/libs/export_metadata.c
+++ b/src/libs/export_metadata.c
@@ -104,27 +104,12 @@ static void _add_selected_metadata(GtkTreeView *view, dt_lib_export_metadata_t *
 }
 
 // choice of a metadata tag
-static gboolean _click_on_metadata_list(GtkWidget *view, GdkEventButton *event, dt_lib_export_metadata_t *d)
+static void _metadata_activated(GtkTreeView *tree_view,
+                                GtkTreePath *path,
+                                GtkTreeViewColumn *column,
+                                dt_lib_export_metadata_t *d)
 {
-  if(event->type == GDK_2BUTTON_PRESS && event->button == 1)
-  {
-
-    GtkTreeSelection *selection = gtk_tree_view_get_selection(GTK_TREE_VIEW(view));
-    GtkTreePath *path = NULL;
-    // Get tree path for row that was clicked
-    if(gtk_tree_view_get_path_at_pos(GTK_TREE_VIEW(view), (gint)event->x, (gint)event->y, &path, NULL, NULL, NULL))
-    {
-      gtk_tree_selection_select_path(selection, path);
-      if(event->type == GDK_2BUTTON_PRESS && event->button == 1)
-      {
-        _add_selected_metadata(GTK_TREE_VIEW(view), d);
-        gtk_tree_path_free(path);
-        return TRUE;
-      }
-    }
-    gtk_tree_path_free(path);
-  }
-  return FALSE;
+  _add_selected_metadata(tree_view, d);
 }
 
 // routine to set individual visibility flag
@@ -160,24 +145,26 @@ static void _tag_name_changed(GtkEntry *entry, dt_lib_export_metadata_t *d)
 // dialog to add metadata tag into the formula list
 static void _add_tag_button_clicked(GtkButton *button, dt_lib_export_metadata_t *d)
 {
-  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("select tag"), GTK_WINDOW(d->dialog), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                       _("add"), GTK_RESPONSE_ACCEPT, _("done"), GTK_RESPONSE_NONE, NULL);
-  g_signal_connect(dialog, "key-press-event", G_CALLBACK(dt_handle_dialog_enter), NULL);
-  gtk_window_set_default_size(GTK_WINDOW(dialog), 300, -1);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("select tag"), GTK_WINDOW(d->dialog),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_add"), GTK_RESPONSE_ACCEPT,
+                                                  _("_done"), GTK_RESPONSE_NONE, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_NONE);
+  gtk_window_set_default_size(GTK_WINDOW(dialog), DT_PIXEL_APPLY_DPI(500), DT_PIXEL_APPLY_DPI(300));
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
   GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 8);
   gtk_container_set_border_width(GTK_CONTAINER(vbox), 8);
-  gtk_container_add(GTK_CONTAINER(area), vbox);
+  gtk_box_pack_start(GTK_BOX(area), vbox, TRUE, TRUE, 0);
 
   GtkWidget *entry = gtk_entry_new();
   d->sel_entry = entry;
   gtk_entry_set_text(GTK_ENTRY(entry), "");
   gtk_widget_set_tooltip_text(entry, _("list filter"));
-  gtk_box_pack_start(GTK_BOX(vbox), entry, TRUE, TRUE, 0);
+  gtk_entry_set_activates_default(GTK_ENTRY(entry), TRUE);
+  gtk_box_pack_start(GTK_BOX(vbox), entry, FALSE, TRUE, 0);
   g_signal_connect(G_OBJECT(entry), "changed", G_CALLBACK(_tag_name_changed), d);
 
   GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
-  gtk_widget_set_size_request(w, DT_PIXEL_APPLY_DPI(500), DT_PIXEL_APPLY_DPI(300));
   gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
   gtk_box_pack_start(GTK_BOX(vbox), w, TRUE, TRUE, 0);
   GtkTreeView *view = GTK_TREE_VIEW(gtk_tree_view_new());
@@ -219,7 +206,7 @@ static void _add_tag_button_clicked(GtkButton *button, dt_lib_export_metadata_t 
   gtk_tree_sortable_set_sort_column_id(GTK_TREE_SORTABLE(liststore), DT_LIB_EXPORT_METADATA_COL_XMP, GTK_SORT_ASCENDING);
   gtk_tree_view_set_model(view, model);
   g_object_unref(model);
-  g_signal_connect(G_OBJECT(view), "button-press-event", G_CALLBACK(_click_on_metadata_list), (gpointer)d);
+  g_signal_connect(G_OBJECT(view), "row-activated", G_CALLBACK(_metadata_activated), d);
 
   #ifdef GDK_WINDOWING_QUARTZ
     dt_osx_disallow_fullscreen(dialog);
@@ -284,18 +271,20 @@ char *dt_lib_export_metadata_configuration_dialog(char *metadata_presets, const 
   dt_lib_export_metadata_t *d = calloc(1, sizeof(dt_lib_export_metadata_t));
 
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("edit metadata exportation"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_ACCEPT, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("edit metadata exportation"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_ACCEPT, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
   dt_gui_dialog_add_help(GTK_DIALOG(dialog), "export_dialog");
 
   d->dialog = dialog;
-  g_signal_connect(dialog, "key-press-event", G_CALLBACK(dt_handle_dialog_enter), NULL);
 
-  gtk_window_set_default_size(GTK_WINDOW(dialog), 300, -1);
+  gtk_window_set_default_size(GTK_WINDOW(dialog), DT_PIXEL_APPLY_DPI(500), -1);
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
 
   GtkWidget *hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
-  gtk_container_add(GTK_CONTAINER(area), hbox);
+  gtk_box_pack_start(GTK_BOX(area), hbox, TRUE, TRUE, 0);
 
   // general info
   GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 8);
@@ -364,7 +353,6 @@ char *dt_lib_export_metadata_configuration_dialog(char *metadata_presets, const 
   gtk_box_pack_start(GTK_BOX(vbox), label, FALSE, TRUE, 0);
 
   GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
-  gtk_widget_set_size_request(w, DT_PIXEL_APPLY_DPI(450), DT_PIXEL_APPLY_DPI(100));
   gtk_widget_set_hexpand(w, TRUE);
   gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
   gtk_box_pack_start(GTK_BOX(vbox), w, TRUE, TRUE, 0);

--- a/src/libs/geotagging.c
+++ b/src/libs/geotagging.c
@@ -806,9 +806,10 @@ static void _preview_gpx_file(GtkWidget *widget, dt_lib_module_t *self)
 {
   dt_lib_geotagging_t *d = (dt_lib_geotagging_t *)self->data;
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons(
-            _("GPX file track segments"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-            _("done"), GTK_RESPONSE_CANCEL, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("GPX file track segments"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_done"), GTK_RESPONSE_CANCEL, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_CANCEL);
 
   gchar *filedir = gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(widget));
   struct dt_gpx_t *gpx = dt_gpx_new(filedir);

--- a/src/libs/import.c
+++ b/src/libs/import.c
@@ -1699,11 +1699,12 @@ static void _import_from_dialog_new(dt_lib_module_t* self)
   dt_lib_import_t *d = (dt_lib_import_t *)self->data;
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
 
-  d->from.dialog = gtk_dialog_new_with_buttons
-    ( _(_import_text[d->import_case]), NULL, GTK_DIALOG_MODAL,
-      _("cancel"), GTK_RESPONSE_CANCEL,
-      _(_import_text[d->import_case]), GTK_RESPONSE_ACCEPT,
-      NULL);
+  d->from.dialog = gtk_dialog_new_with_buttons(_(_import_text[d->import_case]), NULL,
+                                               GTK_DIALOG_MODAL,
+                                               _("_cancel"), GTK_RESPONSE_CANCEL,
+                                               _(_import_text[d->import_case]), GTK_RESPONSE_ACCEPT,
+                                               NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(d->from.dialog), GTK_RESPONSE_ACCEPT);
   dt_gui_dialog_add_help(GTK_DIALOG(d->from.dialog), "import_dialog");
 
 #ifdef GDK_WINDOWING_QUARTZ
@@ -1808,6 +1809,9 @@ static void _import_from_dialog_new(dt_lib_module_t* self)
   {
     gtk_widget_show_all(d->from.dialog);
   }
+
+  // make sure no buttons focused, so default button is marked
+  gtk_window_set_focus(GTK_WINDOW(d->from.dialog), NULL);
 }
 
 static void _import_set_collection(const char *dirname)
@@ -1975,7 +1979,7 @@ static void _lib_import_from_callback(GtkWidget *widget, dt_lib_module_t* self)
                "\nfurther information can be found in the darktable manual."
                "\n\ninspect darktable preferences -> import."
                "\ncheck and possibly correct the 'base directory naming pattern'"),
-            _("show this information again"), _("understood & done"));
+            _("_show this information again"), _("_understood & done"));
       if(understood)
         dt_conf_set_bool("setup_import_directory", TRUE);
       else

--- a/src/libs/metadata.c
+++ b/src/libs/metadata.c
@@ -461,10 +461,12 @@ static void _menuitem_preferences(GtkMenuItem *menuitem,
                                   dt_lib_module_t *self)
 {
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons
-    (_("metadata settings"), GTK_WINDOW(win),
-     GTK_DIALOG_DESTROY_WITH_PARENT, _("default"), GTK_RESPONSE_YES,
-     _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_ACCEPT, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("metadata settings"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_default"), GTK_RESPONSE_YES,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_ACCEPT, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
   dt_gui_dialog_add_help(GTK_DIALOG(dialog), "metadata_preferences");
   g_signal_connect(dialog, "key-press-event", G_CALLBACK(dt_handle_dialog_enter), NULL);
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
@@ -519,6 +521,9 @@ static void _menuitem_preferences(GtkMenuItem *menuitem,
     (_("visible"), renderer,
      "active", DT_METADATA_PREF_COL_VISIBLE, NULL);
   gtk_tree_view_append_column(GTK_TREE_VIEW(view), column);
+  GtkTreePath *first = gtk_tree_path_new_first ();
+  gtk_tree_view_set_cursor(GTK_TREE_VIEW(view), first, column, FALSE);
+  gtk_tree_path_free(first);
   GtkWidget *header = gtk_tree_view_column_get_button(column);
   gtk_widget_set_tooltip_text(header,
                 _("tick if the corresponding metadata is of interest for you"

--- a/src/libs/metadata_view.c
+++ b/src/libs/metadata_view.c
@@ -1201,8 +1201,11 @@ void _menuitem_preferences(GtkMenuItem *menuitem, dt_lib_module_t *self)
 
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
   GtkWidget *dialog = gtk_dialog_new_with_buttons(_("metadata settings"), GTK_WINDOW(win),
-                                       GTK_DIALOG_DESTROY_WITH_PARENT, _("default"), GTK_RESPONSE_YES,
-                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_ACCEPT, NULL);
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_default"), GTK_RESPONSE_YES,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_ACCEPT, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
   g_signal_connect(dialog, "key-press-event", G_CALLBACK(dt_handle_dialog_enter), NULL);
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
 
@@ -1248,6 +1251,9 @@ void _menuitem_preferences(GtkMenuItem *menuitem, dt_lib_module_t *self)
   column = gtk_tree_view_column_new_with_attributes(_("visible"), renderer,
                                                     "active", DT_METADATA_PREF_COL_VISIBLE, NULL);
   gtk_tree_view_append_column(GTK_TREE_VIEW(view), column);
+  GtkTreePath *first = gtk_tree_path_new_first ();
+  gtk_tree_view_set_cursor(GTK_TREE_VIEW(view), first, column, FALSE);
+  gtk_tree_path_free(first);
 
   // drag & drop
   gtk_tree_view_set_reorderable(GTK_TREE_VIEW(view), TRUE);

--- a/src/libs/modulegroups.c
+++ b/src/libs/modulegroups.c
@@ -3400,33 +3400,30 @@ static void _preset_autoapply_changed(dt_gui_presets_edit_dialog_t *g)
   d->editor_reset = FALSE;
 }
 
-static void _manage_editor_preset_name_verify(GtkWidget *tb, dt_lib_module_t *self)
+static void _manage_editor_preset_name_verify(GtkWidget *tb,
+                                              gpointer params[])
 {
-  GList *names = (GList *)g_object_get_data(G_OBJECT(tb), "existing_names");
-  GtkWidget *lb = (GtkWidget *)g_object_get_data(G_OBJECT(tb), "existing_label");
-  GtkWidget *bt_ok = (GtkWidget *)g_object_get_data(G_OBJECT(tb), "ok_btn");
+  GtkDialog *dialog = params[0];
+  GList *names = params[1];
+  GtkWidget *warning_label = params[2];
+
   const gchar *txt = gtk_entry_get_text(GTK_ENTRY(tb));
+  gboolean good = *txt;
 
   // we don't want empty name
-  if(!g_strcmp0(txt, ""))
+  if(good)
   {
-    gtk_widget_set_visible(lb, TRUE);
-    gtk_widget_set_sensitive(bt_ok, FALSE);
-    return;
-  }
-
-  for(const GList *l = names; l; l = g_list_next(l))
-  {
-    gchar *tx = (gchar *)l->data;
-    if(!g_strcmp0(tx, txt))
+    for(const GList *l = names; l; l = g_list_next(l))
     {
-      gtk_widget_set_visible(lb, TRUE);
-      gtk_widget_set_sensitive(bt_ok, FALSE);
-      return;
+      if(!g_strcmp0(l->data, txt))
+      {
+        good = FALSE;
+        break;
+      }
     }
   }
-  gtk_widget_set_visible(lb, FALSE);
-  gtk_widget_set_sensitive(bt_ok, TRUE);
+  gtk_widget_set_visible(warning_label, !good);
+  gtk_dialog_set_response_sensitive(dialog, GTK_RESPONSE_OK, good);
 }
 
 static void _manage_editor_preset_action(GtkWidget *btn, dt_lib_module_t *self)
@@ -3464,26 +3461,26 @@ static void _manage_editor_preset_action(GtkWidget *btn, dt_lib_module_t *self)
   sqlite3_finalize(stmt);
 
   gint res = GTK_RESPONSE_OK;
-  GtkWidget *dialog
-      = gtk_dialog_new_with_buttons(_("rename preset"), GTK_WINDOW(d->dialog), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                    _("cancel"), GTK_RESPONSE_CANCEL, _("rename"), GTK_RESPONSE_OK, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("rename preset"), GTK_WINDOW(d->dialog),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_cancel"), GTK_RESPONSE_CANCEL,
+                                                  _("_rename"), GTK_RESPONSE_OK, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_OK);
 #ifdef GDK_WINDOWING_QUARTZ
   dt_osx_disallow_fullscreen(dialog);
 #endif
-  GtkWidget *bt_ok = gtk_dialog_get_widget_for_response(GTK_DIALOG(dialog), GTK_RESPONSE_OK);
   GtkWidget *content_area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
   gtk_box_pack_start(GTK_BOX(content_area), gtk_label_new(_("new preset name:")), FALSE, TRUE, 0);
   GtkWidget *lb = gtk_label_new(_("a preset with this name already exists!"));
   GtkWidget *tb = gtk_entry_new();
-  gtk_entry_set_text(GTK_ENTRY(tb), new_name);
-  g_object_set_data(G_OBJECT(tb), "existing_names", names);
-  g_object_set_data(G_OBJECT(tb), "existing_label", lb);
-  g_object_set_data(G_OBJECT(tb), "ok_btn", bt_ok);
-  g_signal_connect(G_OBJECT(tb), "changed", G_CALLBACK(_manage_editor_preset_name_verify), self);
+  gtk_entry_set_activates_default(GTK_ENTRY(tb), TRUE);
+  gtk_entry_set_width_chars(GTK_ENTRY(tb), 10 + g_utf8_strlen(gtk_window_get_title(GTK_WINDOW(dialog)), -1));
+  gpointer verify_params[] = {dialog, names, lb};
+  g_signal_connect(G_OBJECT(tb), "changed", G_CALLBACK(_manage_editor_preset_name_verify), verify_params);
   gtk_box_pack_start(GTK_BOX(content_area), tb, FALSE, TRUE, 0);
   gtk_box_pack_start(GTK_BOX(content_area), lb, FALSE, TRUE, 0);
   gtk_widget_show_all(content_area);
-  _manage_editor_preset_name_verify(tb, self);
+  gtk_entry_set_text(GTK_ENTRY(tb), new_name);
   res = gtk_dialog_run(GTK_DIALOG(dialog));
 
   g_free(new_name);

--- a/src/libs/recentcollect.c
+++ b/src/libs/recentcollect.c
@@ -224,8 +224,10 @@ void _menuitem_preferences(GtkMenuItem *menuitem, dt_lib_module_t *self)
   char confname[200];
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
   GtkWidget *dialog = gtk_dialog_new_with_buttons(_("recent collections settings"), GTK_WINDOW(win),
-                                                  GTK_DIALOG_DESTROY_WITH_PARENT, _("cancel"), GTK_RESPONSE_NONE,
-                                                  _("save"), GTK_RESPONSE_ACCEPT, NULL);
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_ACCEPT, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
   dt_prefs_init_dialog_recentcollect(dialog);
   g_signal_connect(dialog, "key-press-event", G_CALLBACK(dt_handle_dialog_enter), NULL);
 

--- a/src/libs/styles.c
+++ b/src/libs/styles.c
@@ -434,10 +434,12 @@ static void _export_clicked(GtkWidget *w, gpointer user_data)
 
           if(dt_conf_get_bool("plugins/lighttable/style/ask_before_delete_style"))
           {
-            GtkWidget *dialog_overwrite_export = gtk_dialog_new_with_buttons(_("overwrite style?"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                _("cancel"), GTK_RESPONSE_CANCEL,
-                _("skip"), GTK_RESPONSE_NONE,
-                _("overwrite"), GTK_RESPONSE_ACCEPT, NULL);
+            GtkWidget *dialog_overwrite_export = gtk_dialog_new_with_buttons(_("overwrite style?"), GTK_WINDOW(win),
+                                                                             GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                                             _("_cancel"), GTK_RESPONSE_CANCEL,
+                                                                             _("_skip"), GTK_RESPONSE_NONE,
+                                                                             _("_overwrite"), GTK_RESPONSE_ACCEPT, NULL);
+            gtk_dialog_set_default_response(GTK_DIALOG(dialog_overwrite_export), GTK_RESPONSE_CANCEL);
 
             // contents for dialog
             GtkWidget *content_area = gtk_dialog_get_content_area(GTK_DIALOG(dialog_overwrite_export));
@@ -617,10 +619,12 @@ static void _import_clicked(GtkWidget *w, gpointer user_data)
           // use security check/option
           if(dt_conf_get_bool("plugins/lighttable/style/ask_before_delete_style"))
           {
-            GtkWidget *dialog_overwrite_import = gtk_dialog_new_with_buttons(_("overwrite style?"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                _("cancel"), GTK_RESPONSE_CANCEL,
-                _("skip"), GTK_RESPONSE_NONE,
-                _("overwrite"), GTK_RESPONSE_ACCEPT, NULL);
+            GtkWidget *dialog_overwrite_import = gtk_dialog_new_with_buttons(_("overwrite style?"), GTK_WINDOW(win),
+                                                                             GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                                             _("_cancel"), GTK_RESPONSE_CANCEL,
+                                                                             _("_skip"), GTK_RESPONSE_NONE,
+                                                                             _("_overwrite"), GTK_RESPONSE_ACCEPT, NULL);
+            gtk_dialog_set_default_response(GTK_DIALOG(dialog_overwrite_import), GTK_RESPONSE_CANCEL);
 
             // contents for dialog
             GtkWidget *content_area = gtk_dialog_get_content_area(GTK_DIALOG(dialog_overwrite_import));

--- a/src/libs/tagging.c
+++ b/src/libs/tagging.c
@@ -1479,8 +1479,11 @@ static void _pop_menu_dictionary_delete_tag(GtkWidget *menuitem, dt_lib_module_t
   if(img_count > 0 || dt_conf_get_bool("plugins/lighttable/tagging/ask_before_delete_tag"))
   {
     GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-    GtkWidget *dialog = gtk_dialog_new_with_buttons(_("delete tag?"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                  _("cancel"), GTK_RESPONSE_NONE, _("delete"), GTK_RESPONSE_YES, NULL);
+    GtkWidget *dialog = gtk_dialog_new_with_buttons(_("delete tag?"), GTK_WINDOW(win),
+                                                    GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                    _("_cancel"), GTK_RESPONSE_NONE,
+                                                    _("_delete"), GTK_RESPONSE_YES, NULL);
+    gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_NONE);
     gtk_window_set_default_size(GTK_WINDOW(dialog), 300, -1);
     GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
     GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 8);
@@ -1567,8 +1570,11 @@ static void _pop_menu_dictionary_delete_node(GtkWidget *menuitem, dt_lib_module_
   if(tag_count == 0) return;
 
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons( _("delete node?"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                _("cancel"), GTK_RESPONSE_NONE, _("delete"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("delete node?"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_delete"), GTK_RESPONSE_YES, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_NONE);
   gtk_window_set_default_size(GTK_WINDOW(dialog), 300, -1);
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
   GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 8);
@@ -1628,6 +1634,13 @@ static void _pop_menu_dictionary_delete_node(GtkWidget *menuitem, dt_lib_module_
   g_free(tagname);
 }
 
+static void _name_changed(GtkEntry *entry,
+                          GtkDialog *dialog)
+{
+  const gchar *name = gtk_entry_get_text(entry);
+  gtk_dialog_set_response_sensitive(dialog, GTK_RESPONSE_YES, name && *name);
+}
+
 // create tag allows the user to create a single tag, which can be an element of the hierarchy or not
 static void _pop_menu_dictionary_create_tag(GtkWidget *menuitem, dt_lib_module_t *self)
 {
@@ -1648,8 +1661,11 @@ static void _pop_menu_dictionary_create_tag(GtkWidget *menuitem, dt_lib_module_t
         DT_LIB_TAGGING_COL_PATH, &path, DT_LIB_TAGGING_COL_ID, &tagid, -1);
 
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("create tag"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("create tag"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_YES, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_YES);
   gtk_window_set_default_size(GTK_WINDOW(dialog), 300, -1);
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
   GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 8);
@@ -1661,6 +1677,9 @@ static void _pop_menu_dictionary_create_tag(GtkWidget *menuitem, dt_lib_module_t
   label = gtk_label_new(_("name: "));
   gtk_box_pack_start(GTK_BOX(box), label, FALSE, TRUE, 0);
   GtkWidget *entry = gtk_entry_new();
+  gtk_entry_set_activates_default(GTK_ENTRY(entry), TRUE);
+  gtk_dialog_set_response_sensitive(GTK_DIALOG(dialog), GTK_RESPONSE_YES, FALSE);
+  g_signal_connect(entry, "changed", G_CALLBACK(_name_changed), dialog);
   gtk_box_pack_end(GTK_BOX(box), entry, TRUE, TRUE, 0);
 
   GtkWidget *category;
@@ -1778,8 +1797,11 @@ static void _pop_menu_dictionary_edit_tag(GtkWidget *menuitem, dt_lib_module_t *
   }
 
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("edit"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("edit"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_YES, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_YES);
   gtk_window_set_default_size(GTK_WINDOW(dialog), 300, -1);
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
   GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 8);
@@ -1808,6 +1830,8 @@ static void _pop_menu_dictionary_edit_tag(GtkWidget *menuitem, dt_lib_module_t *
   label = gtk_label_new(_("name: "));
   gtk_box_pack_start(GTK_BOX(box), label, FALSE, TRUE, 0);
   GtkWidget *entry = gtk_entry_new();
+  gtk_entry_set_activates_default(GTK_ENTRY(entry), TRUE);
+  g_signal_connect(entry, "changed", G_CALLBACK(_name_changed), dialog);
   gtk_entry_set_text(GTK_ENTRY(entry), subtag ? subtag : tagname);
   gtk_box_pack_end(GTK_BOX(box), entry, TRUE, TRUE, 0);
 
@@ -2052,8 +2076,11 @@ static void _pop_menu_dictionary_change_path(GtkWidget *menuitem, dt_lib_module_
   if(tag_count == 0) return;
 
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
-  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("change path"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
-                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("change path"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_YES, NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_YES);
   gtk_window_set_default_size(GTK_WINDOW(dialog), 300, -1);
   GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
   GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 8);
@@ -2078,6 +2105,7 @@ static void _pop_menu_dictionary_change_path(GtkWidget *menuitem, dt_lib_module_
   g_free(text);
 
   GtkWidget *entry = gtk_entry_new();
+  gtk_entry_set_activates_default(GTK_ENTRY(entry), TRUE);
   gtk_entry_set_text(GTK_ENTRY(entry), tagname);
   gtk_box_pack_start(GTK_BOX(vbox), entry, FALSE, TRUE, 0);
 
@@ -3467,9 +3495,10 @@ void _menuitem_preferences(GtkMenuItem *menuitem, dt_lib_module_t *self)
   GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
   GtkWidget *dialog = gtk_dialog_new_with_buttons(_("tagging settings"), GTK_WINDOW(win),
                                                   GTK_DIALOG_DESTROY_WITH_PARENT,
-                                                 _("cancel"), GTK_RESPONSE_NONE,
-                                                 _("save"), GTK_RESPONSE_ACCEPT, NULL);
+                                                  _("_cancel"), GTK_RESPONSE_NONE,
+                                                  _("_save"), GTK_RESPONSE_ACCEPT, NULL);
   g_signal_connect(dialog, "key-press-event", G_CALLBACK(dt_handle_dialog_enter), NULL);
+  gtk_dialog_set_default_response(GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT);
   dt_prefs_init_dialog_tagging(dialog);
 
 #ifdef GDK_WINDOWING_QUARTZ

